src/main/java/com/minecraftabnormals/abnormals_core/core/util/DataUtil.java

```diff
@@ -314,4 +301,89 @@ public static void registerConfigPredicate(IConfigPredicateSerializer<?> seriali
+	 * @param behavior The {@link IDispenseItemBehavior} that will be used if the {@code condition} is met.
+	 *
+	 * @see DataUtil#registerAlternativeDispenseBehavior(String, Item, BiPredicate, IDispenseItemBehavior, Function)
+	 *
+	 * @author abigailfails
+	 */
+	public static void registerAlternativeDispenseBehavior(String modId, Item item, BiPredicate<IBlockSource, ItemStack> condition, IDispenseItemBehavior behavior) {
+		registerAlternativeDispenseBehavior(modId, item, condition, behavior, id -> 0);
+	}
+
+	/**
+	 * <p>Registers a {@link IDispenseItemBehavior} that will perform the new behavior if the condition is met and the behavior that was already in the registry if not.
+	 * This works even if multiple mods add new behavior to the same item, though it is possible that the conditions
+	 * might overlap, which is what {@code modComparator} is intended to solve.</p>
+	 *
+	 * <p>For example, if a mod with the ID {@code a} has a behavior where its condition passes if any block is in front
+	 * of the dispenser, but a mod with the ID {@code b} has a behavior for the same item that passes only if a specific
+	 * block is in front of the dispenser, authors may want to make sure that {@code b}'s condition is registered after
+	 * {@code a}'s. In this case, {@code a}'s {@code modComparator} should return -1 if the string given by the function
+	 * equals {@code b}, and {@code b}'s should return 1 if the string equals {@code a}.</p>
+	 *
+	 * <p>Ideally, the condition should be implemented such that the predicate only passes if the new behavior will be 'successful', avoiding problems with failure sounds not playing.</p>
+	 *
+	 * <p>Since Abnormals Core handles registering the condition at the right time, mods should call this method as early as possible,
+	 * such as in a {@code static{}} block.</p>
+	 *
+	 * @param item The {@link Item} to register the {@code newBehavior} for.
+	 * @param condition A {@link BiPredicate} that takes in {@link IBlockSource} and {@link ItemStack} arguments, returning true if the {@code newBehavior} should be performed.
+	 * @param behavior The {@link IDispenseItemBehavior} that will be used if the {@code condition} is met.
+	 * @param modComparator A {@link Function} that inputs a String representing the mod id for another behavior registered to the same item.
+	 *                      It should return 1 if {@code behavior} is to be registered after the other behavior, -1 if
+	 *                      it should go before, and 0 in any other case.
+	 *
+	 * @author abigailfails
+	 */
+	public static void registerAlternativeDispenseBehavior(String modId, Item item, BiPredicate<IBlockSource, ItemStack> condition, IDispenseItemBehavior behavior, Function<String, Integer> modComparator) {
+		ALTERNATIVE_DISPENSE_BEHAVIORS.add(new AlternativeDispenseBehavior(modId, item, condition, behavior, modComparator));
+	}
```

Comment:
Only one new method here is necessary.
`AlternativeDispenseBehavior` holds all the information needed to add an alternative behavior meaning there should be one method that takes in an `AlternativeDispenseBehavior` instance.
Doing this would also mean people could add `AlternativeDispenseBehavior` extensions to the list, allowing more freedom to the sorting.

Reply:
Done in latest commit.